src/lib.rs

```diff
@@ -113,6 +124,10 @@ pub enum ConfyError {
     #[error("Bad TOML data")]
     BadTomlData(#[source] toml::de::Error),
 
+    #[cfg(feature = "basic_toml_conf")]
+    #[error("Bad TOML data")]
+    BadTomlData(#[source] basic_toml::Error),
+
```

Comment:
I think we might want to make an abstraction layer to make this a bit better. For the error types, we can probably make our own and import what makes sense depending on the `toml` crate used. 

For the functions below we can probably do some import `cfg` gate to reduce the differences here e.g. 

```rust
#[cfg(feature = "toml_conf")]
use toml::to_string as toml_to_string;
#[cfg(feature = "basic_toml_conf")]
use base_toml::to_string as toml_to_string;
```

As well as the function names below, so we can just use `#[cfg(any(feature = "basic_toml_conf", feature = "toml_conf")]